### PR TITLE
Added markdown-it with support for emojis, permalinks, and checkboxes

### PR DESCRIPTION
**Pull request by niftylettuce**

### CODE REVIEW

**Comment by shime** [comment]
Can you fix those Travis issues?

**Comment by shime** [comment]
Permalinks are not styled correctly, can we copy Github's styling of it?

**Comment by niftylettuce** [comment]
@shime not sure how to fix them - I believe the errors are related to the other issue #16 perhaps?  are you able to figure it out? Also I looked at GitHub's styling and opened an issue at <https://github.com/valeriangalliat/markdown-it-anchor/issues/28> - but we could copy GH's styling through the options here <https://github.com/valeriangalliat/markdown-it-anchor/#usage>.

**Comment by niftylettuce** [comment]
@shime Please accept this pull request - I've updated it, upgraded dependencies, and fixed the tests.  I also removed the Permalink styling since it didn't match GitHub exactly -- so perhaps once the issue at <https://github.com/valeriangalliat/markdown-it-anchor/issues/28> is resolved and figured out we can add this to our project.  Should be good to go.  Of course once you merge you'll need to version bump in `package.json` and then npm publish a new version.  If you'd like to have me as a collaborator, I'd love to - and I'm on NPM as niftylettuce as well.

**Comment by niftylettuce** [comment]
Also FWIW you might want to do a major semver version bump, I only have support in this for Node v6.x to Node v8.x per `.travis.yml` changes.

**Comment by niftylettuce** [comment]
Per my findings at <https://github.com/assaf/zombie/issues/1039>, I also had to modify the test script to not use `Browser.localhost`. Instead we pass the complete URL to `browser.visit`. See https://github.com/adonisjs/adonis-bodyparser/commit/31768c1d78b7122331778e2ea79a158374df6012 as well.  Pushing this shortly, then build will be fixed.

**Comment by niftylettuce** [comment]
Actually, I think we can get GH headers in - I just found this - https://github.com/Flet/markdown-it-github-headings - let me try it.

**Comment by niftylettuce** [comment]
@shime Done. Here's an example GIF for you 🎉 

gif

:shipit:

**Comment by shime** [comment]
Wow! This looks awesome. Can you fix those extra semicolon issues on Travis?

**Comment by niftylettuce** [comment]
@shime Fixed!  See my latest push.  Builds should pass, oops 😆

File: server.js [hunk 1]
```diff
@@ -19,17 +32,19 @@ module.exports = function (opts) {
...
-var ImageExtesions = 'jpg jpeg gif png svg bmp xbm'.split(' ')
+var ImageExtensions = 'jpg jpeg gif png svg bmp xbm'.split(' ')
 
 function Server (opts) {
   opts = opts || {}
 
-  this.port = opts.port || 1337
+  var self = this
+
+  this.port = opts.port || 3030
```
Comment (shime) [inline server.js]: Can you please change the default port back to `1337`? Thanks!

Comment (niftylettuce) [inline server.js]: One moment

**Comment by niftylettuce** [comment]
Ah there's another issue with tests now that GH is in, one moment.

**Comment by shime** [comment]
Thank you so much, this is great! I added you as a collaborator on Github and NPM. Once you fix the tests feel free to merge it and push the new version.

**Comment by niftylettuce** [comment]
Ah, I figured out why port 1337 wasn't working for me - it's due to Tunnelblick per <https://tunnelblick.net/cz33Prefs.html>... Haha.


Fixing so now..

**Comment by niftylettuce** [comment]
@shime Just pushed final version! Once this build finishes with travis you should be good to go - release new tag/version bump package.json then npm publish.  If any issues arise just lmk!

**Comment by niftylettuce** [comment]
Oh sweet, I'll publish it - thanks for access.

**Comment by niftylettuce** [comment]
@shime Version 2.0.0 is released!

**Comment by shime** [comment]
🎉🎊

**Pull request merged**